libtransmission/bandwidth.cc

```diff
@@ -288,97 +288,94 @@ void tr_bandwidthAllocate(tr_bandwidth* b, tr_direction dir, unsigned int period
     tr_ptrArrayDestruct(&tmp, nullptr);
 }
 
-void tr_bandwidthSetPeer(tr_bandwidth* b, tr_peerIo* peer)
+void tr_bandwidth::setPeer(tr_peerIo* peer)
 {
-    TR_ASSERT(tr_isBandwidth(b));
+    TR_ASSERT(tr_isBandwidth(this));
     TR_ASSERT(peer == nullptr || tr_isPeerIo(peer));
 
-    b->peer = peer;
+    this->peer = peer;
 }
```

Comment:
- This is a pretty trivial function; IMO it could be added to the header.

- Also, `TR_ASSERT(tr_isBandwidth())` probably isn't as necessary inside member functions, since C++ managed to resolve the member function. We ought to remove that style of assertions as these functions become member functions.